codecov.yaml

```diff
@@ -0,0 +1,33 @@
+codecov:
+  notify:
+    require_ci_to_pass: no
+
+coverage:
+  precision: 2
+  round: down
+  range: "50...100"
+
+  status:
+    project:
+      default:
+        # allow the project's overall coverage to drop by X%
```

Comment:
This may be problematic. Do we want to be blocking PRs that drop unit test coverage by 1%? I think not.

Reply:
@kargakis so what do you suggest we put here? I could also block the project check by setting `project: no`. In that case only the `patch` check (for PRs) will be performed. And right now that check is configured to pass if and only if the changes introduced in the PR have a coverage ratio of at least 50%

Reply:
I see codecov as more of a reporting tool rather than something that would block PRs in CI so I am not really sure this setting makes sense at all but that's just my opinion. May be worth to bring it up during scrum, I hadn't thought before we were going to be blocking PRs with this and it's probably unexpected (devs writting unit tests in order to make their PRs merge is smelly).

Reply:
Thats also fine. We could disable the codecov checks altogether and depend only the PR report comments by setting:

```
status:
    project: no
    patch: no
    changes: no
```

Lets discuss that during scrum